### PR TITLE
Add new method for authentication and support multiple approve

### PR DESCRIPTION
**Pull request by NNagato**

### CODE REVIEW

**Comment by secmask** [comment]
@halink0803 could you help to review this?

**Comment by halink0803** [comment]
@secmask yes, I can.

File: reservesetting/http/setting_change.go [hunk 1]
```diff
@@ -304,14 +304,54 @@ func (s *Server) rejectSettingChange(c *gin.Context) {
...
+		input struct {
+			ID uint64 `uri:"id" binding:"required"`
+		}
+	)
+	if err := c.ShouldBindUri(&input); err != nil {
+		httputil.ResponseFailure(c, httputil.WithError(err))
+		return
+	}
+	keyID := s.getKeyIDFromContext(c)
+	if err := s.storage.DispproveSettingChange(keyID, input.ID); err != nil {
```
Comment (secmask) [inline reservesetting/http/setting_change.go:322]: there's a typo in `DisapproveSettingChange`

File: gateway/cmd/config.json.example [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+  "read_role": [
+    {
+      "AccessKeyID": "read_access_key_id",
```
Comment (secmask) [inline gateway/cmd/config.json.example:4]: it's not consistent to use uppercase here, we can redefine KeyPair struct with json tag, I would suggest also add optional `name` field for keyPair so Gary can easy to remember who this key belong to.

File: reservesetting/http/setting_change.go (hunk 1 above)
Comment (NNagato) [reply reservesetting/http/setting_change.go]: fixed in https://github.com/KyberNetwork/reserve-data/pull/1207/commits/15adffd235ddfc95d487636c639e79750465eee4

File: gateway/cmd/config.json.example (hunk 2 above)
Comment (NNagato) [reply gateway/cmd/config.json.example]: updated in https://github.com/KyberNetwork/reserve-data/pull/1207/commits/15adffd235ddfc95d487636c639e79750465eee4

**Review by secmask — APPROVED** [review]

**Pull request merged**